src/ui/public/vislib/vis.js

```diff
@@ -25,12 +25,12 @@ export default function VisFactory(Private) {
    * @param config {Object} Parameters that define the chart type and chart options
    */
   class Vis extends Events {
-    constructor($el, config) {
+    constructor($el, visConfigArgs) {
       super(arguments);
       this.el = $el.get ? $el.get(0) : $el;
       this.binder = new Binder();
-      this._attr = config;
-      this._attr.el = this.el;
+      this.visConfigArgs = visConfigArgs;
```

Comment:
and to instantiate visConfig we need data which is only availible in render method ... 
after adding hasLegend the only reason we need this is to store it for later (in render method) when data is available.